src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java

```diff
@@ -202,13 +201,10 @@ protected void updateCitizens()
         //Removes all citizens which already have a job.
         citizens = colony.getCitizens().values().stream()
                      .filter(citizen -> !citizen.isChild())
-                     .filter(citizen -> {
-                         return (citizen.getWorkBuilding() == null)
-                                 || building.getPosition().equals(citizen.getWorkBuilding())
-                                 || colony.getBuilding(citizen.getWorkBuilding()).getSchematicName().contains("library")
-                                 || colony.getBuilding(citizen.getWorkBuilding()).getSchematicName().contains("archery")
-                                 || colony.getBuilding(citizen.getWorkBuilding()).getSchematicName().contains("combatacademy");
-                     }).sorted(Comparator.comparing(ICitizenDataView::getName))
+                     .filter(citizen -> (citizen.getWorkBuilding() == null)
+                             || building.getPosition().equals(citizen.getWorkBuilding())
+                             || colony.getBuilding(citizen.getWorkBuilding()) instanceof IBuildingCanBeHiredFrom).
```

Comment:
So @gattsuru, this is what I mean by isn't working, we've changed how that check is implemented to this, but when the view is generated, I'm not sure that it is getting the interface.